woocommerce/woocommerce-ios
# [Tooling] Update Xcode version refs in tooling scripts

## Why?

This updates the references to the minimum Xcode version needed to compile WCiOS to `14.0`, as some references in `.xcversion` and `Fastfile` were still mentioning `~> 13.4` being the minimum, while the project now requiring Xcode 14 minimum to compile.

## Context

While I was trying to [run the Unit Tests in my Xcode on my local machine](https://github.com/woocommerce/woocommerce-ios/pull/8025#pullrequestreview-1168395779), I encountered some issues because:

1. I hadn't worked on this repo in a while, so hadn't compiled the code in ages
2. I hadn't have to use Xcode in a while on that machine either, so my default Xcode (`xcode-select -p`) was still 13.4.1

As a result, I accidentally tried to run the Unit Tests with Xcode 13.4, which failed to compile — at the very least because of [this line](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Networking/Networking/Remote/Remote.swift#L221) which uses [a new feature only available since Swift 5.7](https://github.com/apple/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md) — and I had to open the project with Xcode 14 to be able to compile it.

[Our CI config for this project is also using Xcode 14](https://github.com/woocommerce/woocommerce-ios/blob/trunk/.buildkite/pipeline.yml#L14) anyway, so this makes sense to keep those aligned.

## To Test

 - Check that CI passes, proving that the checks made via `xcversion()` calls in the `Fastfile` as working and relying on `.xcode-version` file as expected
 - Run `bundle exec fastlane build_for_testing` locally and validate that it passes (i.e. it doesn't complain about Xcode 14 missing).
    - Ideally, it would be nice to test this while you _don't_ have Xcode `14.0` installed on your machine, but do have Xcode `14.1`, so that we can prove that the `~>` operator works as expected and does not restrict to _just_ `14.0` exact match. (I personally only had Xcode 14.0 on my machine, and have wifi trouble today so not the day I'm going to download `14.1` to test 😅 , but hopefully someone can test on their machine)
 - Temporarily change `.xcode-version` to something like `~> 14.5`, try `bundle exec fastlane build_for_testing` again, and validate it complains about Xcode 14.5 or later not being found on your machine
     - _(the proper test for that would instead be to Zip your Xcode 14 `.app` and delete that `.app` so that it couldn't be found on your Mac anymore and retry to verify it complains about not finding Xcode 14 being installed anymore… but that might take a bit long to zip / uninstall Xcode 14 then unzip/reinstall it once done, so…)_
